File: 002/problem.go
package main

import (
    "fmt"
)

func main() {
    fmt.Println(solve_problem())
}

func solve_problem() (sum int) {
    sum = 0

    for a, b := 1, 2; a <= 4e6; a, b = b, a+b {
        if b%2 == 0 {
            sum += b
        }
    }

    return
}

File: 002/problem.php
<?php

echo solve_problem(), "\n";

function solve_problem() {
    $sum = 0;
    $current = 1;
    $next = 2;

    while($current <= 4e6) {
        $new = $current + $next;
        if($next % 2 == 0) {
            $sum += $next;
        }
        $current = $next;
        $next = $new;
    }

    return $sum;
}
